lumicks/pylake/confocal.py

```diff
@@ -55,15 +55,25 @@ def from_dataset(cls, h5py_dset, file):
         return cls(name, file, start, stop, json_data)
 
     def _get_photon_count(self, name):
-        photon_count = getattr(self.file, f"{name}_photon_count".lower())[self.start:self.stop]
+        timeline_start = self.file.h5["Photon count"][name].attrs["Start time (ns)"]
+        timeline_sample_rate = self.file.h5["Photon count"][name].attrs["Sample rate (Hz)"]
+        timeline_dt = int(1e9 /timeline_sample_rate)
```

Comment:
Is there a reason you go with the raw `h5` here? I'd prefer using the dt from the higher level API.

Reply:
My thought was to avoid grabbing the `photon_count` slice twice, because as you mentioned before it's a bit confusing and looks like duplicated code. I also thought, performance wise, it was somewhat wasteful to construct the whole slice just to grab the `start` and `dt`, but thinking about it more I guess it's not a particularly complicated operation.

I'm fine with going back to the original version, but also perhaps would it be nice to have a small helper method to grab the time info without constructing the whole slice? Would this be useful elsewhere maybe?

Reply:
Yeah, I see what you mean. Unfortunately, for the sted workaround, you need the first timestamp after or equal to `Scan.start`, so I don't think there's a nice way around the slicing it already. `Continuous.slice` is actually fairly lightweight, it doesn't construct all the timestamps to just give you the start.